recipes/wt/all/conanfile.py

```diff
@@ -34,52 +36,79 @@ class WtConan(ConanFile):
         "connector_fcgi": [True, False]
     }
     default_options = {
-        'shared': False,
-        'fPIC': True,
-        'with_ssl': True,
-        'with_sqlite': True,
-        'with_postgres': True,
-        'with_mysql': True,
-        'with_mssql': False,
-        'with_test': False,
-        'with_dbo': True,
-        'with_opengl': False,
-        'with_unwind': True,
-        'no_std_locale': False,
-        'no_std_wstring': False,
-        'multi_threaded': True,
-        'connector_http': True,
-        'connector_isapi': True,
-        'connector_fcgi': False
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": True,
+        "with_sqlite": True,
+        "with_postgres": True,
+        "with_mysql": True,
+        "with_mssql": False,
+        "with_test": False,
+        "with_dbo": True,
+        "with_opengl": False,
+        "with_unwind": True,
+        "no_std_locale": False,
+        "no_std_wstring": False,
+        "multi_threaded": True,
+        "connector_http": True,
+        "connector_isapi": True,
+        "connector_fcgi": False
         }
 
-    _source_subfolder = "source_subfolder"
-    _build_subfolder = "build_subfolder"
-
     _cmake = None
 
-    requires = ('zlib/1.2.11', 'boost/1.73.0')
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
-    def requirements(self):
-        if self.options.with_ssl:
-            self.requires('openssl/1.1.1g')
-        if self.options.with_sqlite:
-            self.requires('sqlite3/3.31.1')
-        if self.options.with_mysql:
-            self.requires('libmysqlclient/8.0.17')
-        if self.options.with_postgres:
-            self.requires('libpq/11.5')
-        if self.options.with_unwind:
-            self.requires('libunwind/1.3.1')
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
             del self.options.connector_fcgi
         else:
             del self.options.connector_isapi
         if self.settings.os not in ["Linux", "FreeBSD"]:
-            self.options.with_unwind = False
+            del self.options.with_unwind
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.options.with_dbo:
+            del self.options.with_sqlite
+            del self.options.with_postgres
+            del self.options.with_mysql
+            del self.options.with_mssql
+        self._strict_options_requirements()
+
+    def _strict_options_requirements(self):
+        self.options["boost"].without_program_options = False
+        self.options["boost"].without_filesystem = False
+        self.options["boost"].without_thread = False
```

Comment:
Should this be moved to `build()` and changed to a `raise ConanInvalidConfiguration` when the options are invalid?

Reply:
`build()` is not always executed.

Reply:
Mmmhh, looking at boost recipe, `header_only` must not be True also

Reply:
> `build()` is not always executed. But checks are also done in build method.

Can you explain this please? I don't see a `package_id` function.
Also, the options of requirements are not available in `configure`. They are always `None`.

Reply:
There is no check in `configure`, boost options are forced.

Options of requirements don't contribute to `package_id` (at least by default), so `build()` might not be triggered if boost options are modified.

Reply:
Thanks, I think we need a new conan feature: `post_requirements` or something similar for checking options of requirements.

Reply:
I agree. There is already a proposal https://github.com/conan-io/conan/issues/7591

Reply:
Maybe add a FIXME/TODO to remember us to fix this later.

Reply:
yep, done